src/Shipment.php

```diff
@@ -0,0 +1,352 @@
+        array $group_ids,
+        array $options = []
+    ) {
+        $this->verifyPostalCode($manifest_address);
+        $content = [
+            'group-ids' => [
+                'group-id' => $group_ids
+            ],
+            'requested-shipping-point' => $manifest_address['address-details']['postal-zip-code'],
+            'cpc-pickup-indicator' => true,
+            'detailed-manifests' => true,
+            'manifest-address' => $manifest_address,
+        ];
+
+        if (!empty($options['option_codes'])) {
+            $content['options']['option'] = $this->parseOptionCodes($options);
+        }
+
+        $xml = Array2XML::createXML('transmit-set', $content);
+        $envelope = $xml->documentElement;
+        $envelope->setAttribute('xmlns',
+            'http://www.canadapost.ca/ws/manifest-v8');
+        $payload = $xml->saveXML();
+
+        $response = $this->post(
+            "rs/{$this->customerNumber}/{$this->customerNumber}/shipment",
+            [
+                'Accept' => 'application/vnd.cpc.manifest-v8+xml',
+                'Content-Type' => 'application/vnd.cpc.manifest-v8+xml',
+            ],
+            $payload,
+            $options
+        );
+        return $response;
+
+    }
+
+    /**
+     * Get the manifest from Canada Post server.
+     *
+     * @param string $manifest_id
+     *   The manifest id.
+     * @param string $extra
+     *   Gets the specific details for a shipment.
+     * @param array $options
+     *   The options array.
+     *
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getManifest($manifest_id, $extra = '', array $options = [])
+    {
+        $response = $this->get(
+            "rs/{$this->config['customer_number']}/{$this->config['customer_number']}/manifest/{$manifest_id}/{$extra}",
+            ['Accept' => 'application/vnd.cpc.manifest-v8+xml'],
+            $options
+        );
+        return $response;
+
+    }
+
+    /**
+     * Get Manifests from Canada Post within the specified range.
+     *
+     * @param string $start
+     *   The beginning range. YmdHs format, eg. 201808282359.
+     * @param string $end
+     *   The end range, defaults to current time. YmdHs format, eg. 201808282359.
+     * @param string $tracking_pin
+     *   The Tracking PIN of the shipment to retrieve.
+     * @param array $options
+     *   The options array.
+     *
+     * @see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/shippingmanifest/manifests.jsf
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getManifests(
+        $start,
+        $end = '',
+        $tracking_pin = '',
+        array $options = []
+    ) {
+        if (empty($end)) {
+            $end = date('YmdHs');
+        }
+        $query_params = "start={$start}&end{$end}";
```

Comment:
Would be better to be consistent and name the arguments `from` and `to` like in the methods above.

Reply:
these are the arguments used by canadapost (which i also find stupid that they named differently) but i figured its better to use the same names that the api use, no?

Reply:
Ok. Up to you really, both approaches have merits. Makes sense to keep Canada Post's terms, but it is also common and totally acceptable to change it if you can provide a more consistent API. At the end of the day the SDK user will only be using the SDK and never the Canada Post directly so there won't be any perceived inconsistency as far as they're concerned.